conda_forge_tick/auto_tick.py

```diff
@@ -177,307 +497,138 @@ def run(
         )
-        eval_cmd(["rm", "-rf", feedstock_dir])
         return False, False
 
-    # rerender, maybe
-    diffed_files: typing.List[str] = []
-    with pushd(feedstock_dir):
-        msg = migration_run_data["commit_message"]
-        try:
-            eval_cmd(["git", "add", "--all", "."])
-            if migrator.allow_empty_commits:
-                eval_cmd(["git", "commit", "--allow-empty", "-am", msg])
-            else:
-                eval_cmd(["git", "commit", "-am", msg])
-        except CalledProcessError as e:
-            logger.info(
-                "could not commit to feedstock - "
-                "likely no changes - error is '%s'" % (repr(e)),
-            )
-            # we bail here if we do not plan to rerender and we wanted an empty
-            # commit
-            # this prevents PRs that don't actually get made from getting marked as done
-            if migrator.allow_empty_commits and not rerender:
-                raise e
-
-        if rerender:
-            head_ref = eval_cmd(["git", "rev-parse", "HEAD"]).strip()
-            logger.info("Rerendering the feedstock")
-
-            try:
-                rerender_msg = rerender_feedstock(feedstock_dir, timeout=900)
-                if rerender_msg is not None:
-                    eval_cmd(["git", "commit", "--allow-empty", "-am", rerender_msg])
-
-                make_rerender_comment = False
-            except Exception as e:
-                # I am trying this bit of code to force these errors
-                # to be surfaced in the logs at the right time.
-                print(f"RERENDER ERROR: {e}", flush=True)
-                if not isinstance(migrator, Version):
-                    raise
-                else:
-                    # for check solvable or automerge, we always raise rerender errors
-                    if get_keys_default(
-                        feedstock_ctx.attrs,
-                        ["conda-forge.yml", "bot", "check_solvable"],
-                        {},
-                        False,
-                    ) or get_keys_default(
-                        feedstock_ctx.attrs,
-                        ["conda-forge.yml", "bot", "automerge"],
-                        {},
-                        False,
-                    ):
-                        raise
-                    else:
-                        make_rerender_comment = True
-
-            # If we tried to run the MigrationYaml and rerender did nothing (we only
-            # bumped the build number and dropped a yaml file in migrations) bail
-            # for instance platform specific migrations
-            gdiff = eval_cmd(
-                ["git", "diff", "--name-only", f"{head_ref.strip()}...HEAD"]
-            )
-
-            diffed_files = [
-                _
-                for _ in gdiff.split()
-                if not (
-                    _.startswith("recipe")
-                    or _.startswith("migrators")
```

Comment:
@isuruf @beckermr While designing test cases, I came across this piece of code that was introduced in #516.

What is the purpose of excluding files from a `migrators` directory when checking if a migration rerender is not a trivial MigrationYaml change (at least this is how I called it in my refactored code)? I am not aware of any feedstocks that have a top-level `migrators` directory. The closest to my understanding of this code is probably the `.ci_support/migrations` (sic!) directory, which is not considered here.

I personally would guess that this is is a mix of horrendously outdated and never-caught typo but please enlighten me!

Reply:
So this does indeed look like a bug in terms of the path, however the purpose of the code is as follows. When we run the bot, we run a series of feedstock cleanup tasks on top of the main migrators. These tasks are called minimigrators or piggy-pack migrations. They make changes throughout the feedstock, including in the recipe. Additionally, the README can change for various reasons that don't involve a full rebuild. So this code is meant to make the bot not issue PRs for trivial changes introduced by the rerender.

Reply:
Yep, that's clear from my side. How should we proceed with the bug then? Remove the `migrators` directory check because we did not need it anyway so far?

Reply:
I removed it in 37809fa since the `migrators` directory check did not do anything until now, so apparently we don't need it.

Reply:
I suspect we need to use `.ci_support/migrations` instead. We can do that in a new PR though.